lib/rules/base.js

```diff
@@ -83,10 +83,10 @@ module.exports = class Base {
     return config;
   }
 
-  getVisitor(env) {
+  async getVisitor(env) {
     let pluginContext = this;
     let visitor = {};
-    let ruleVisitor = this.visitor(env);
+    let ruleVisitor = await Promise.resolve(this.visitor(env));
```

Comment:
No need for `Promise.resolve` here, `await {}` (e.g. await on a non-promise) works just fine